exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/DrillSqlWorker.java

```diff
@@ -69,29 +72,43 @@ public static PhysicalPlan getPlan(QueryContext context, String sql, Pointer<Str
     final SqlHandlerConfig config = new SqlHandlerConfig(context, parser);
 
     switch(sqlNode.getKind()){
-    case EXPLAIN:
-      handler = new ExplainHandler(config, textPlan);
-      break;
-    case SET_OPTION:
-      handler = new SetOptionHandler(context);
-      break;
-    case OTHER:
-      if(sqlNode instanceof SqlCreateTable) {
-        handler = ((DrillSqlCall)sqlNode).getSqlHandler(config, textPlan);
+      case EXPLAIN:
+        handler = new ExplainHandler(config, textPlan);
         break;
-      }
-
-      if (sqlNode instanceof DrillSqlCall) {
-        handler = ((DrillSqlCall)sqlNode).getSqlHandler(config);
+      case SET_OPTION:
+        handler = new SetOptionHandler(context);
         break;
-      }
-      // fallthrough
-    default:
-      handler = new DefaultSqlHandler(config, textPlan);
+      case OTHER:
+        if(sqlNode instanceof SqlCreateTable) {
+          handler = ((DrillSqlCall)sqlNode).getSqlHandler(config, textPlan);
+          break;
+        }
+
+        if (sqlNode instanceof DrillSqlCall) {
+          handler = ((DrillSqlCall)sqlNode).getSqlHandler(config);
+          break;
+        }
+        // fallthrough
+      default:
+        handler = new DefaultSqlHandler(config, textPlan);
     }
 
     try {
-      return handler.getPlan(sqlNode);
+      try {
+        return handler.getPlan(sqlNode);
+      } catch (UserException e) {
+        if (context.getOption(ExecConstants.DYNAMIC_UDF_SUPPORT_ENABLED).bool_val) {
+          final Throwable rootCause = ExceptionUtils.getRootCause(e);
+          if (rootCause instanceof SqlValidatorException
+              && StringUtils.contains(rootCause.getMessage(), "No match found for function signature")) {
+            if (context.getFunctionRegistry().loadRemoteFunctions()) {
```

Comment:
Makes sense, I'll think how to do add such approach in existing code. May be I'll introduce local registry version.